presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -913,10 +918,11 @@ private Scope legacyVisitQuerySpecification(QuerySpecification node, Optional<Sc
             // to pass down to analyzeFrom
 
             Scope sourceScope = analyzeFrom(node, scope);
+            Scope sourceScopeWithWindow = createWindowScope(node, Optional.of(sourceScope));
```

Comment:
Quoting from the specs `ISO/IEC 9075-2:2011(E) 7.11 12)`:

> ```The scope of the <new window name> simply contained in WDEF consists of any <window definition>s that follow WDEF in the <window clause>, together with the <select list> of the <query specification> or <select statement: single row> that simply contains the <window clause>. If the <window clause> is simply contained in a <query specification> that is the <query expression body> of a <query expression> that is a simple table query, then the scope of <new window name> also includes the <order by clause>, if any, of that <query expression>.```

My interpretation of the above is that the scope of window definitions includes only the `SELECT` statement and `<order by>` clause, if present. It would be wrong to pass window scope into analyzeFrom. 

For ex., for a query such as this: `SELECT SUM(s) OVER w FROM ( SELECT orderstatus, orderkey, SUM(orderkey) OVER w s FROM orders WINDOW w AS (ORDER BY orderstatus, orderkey) ) WINDOW w AS (PARTITION BY orderstatus);`, the definition of `w` is different in the outer query and the subquery; the window scope of outer query shouldn't be passed to the subquery.